landscape/client/manager/scriptexecution.py

```diff
@@ -105,6 +106,11 @@ def _run_script(self, filename, uid, gid, path, env, time_limit):
             uid = None
         if gid == os.getgid():
             gid = None
+        env = {
+            key: value.encode(sys.getfilesystemencoding(), errors='replace')
```

Comment:
Shouldn't this be using sys.getdefaultencoding()? Filesystem encoding is could similar but that doesn't sound right.

Reply:
That's what I originally thought, but Python 3 uses `sys.getfilesystemencoding` for its [os.environ implementation](https://docs.python.org/3/library/os.html#os.environ) (the unencoded bytes-variant of the environment is os.environb in Python 3). I guess it makes a certain amount of sense as the most important vars in the environment are usually file-system related (PATH etc.). There's also the aspect that `sys.getdefaultencoding()` on Python 2 simply returns 'ascii' regardless of the environment.

Reply:
fair enough. just drop a comment for future me to remember that. :)